### ISSUE TITLE
trigger_heal does not heal unpowered buildables

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#1948]
trigger_heal should repair buildables even if no RC/OM around. This is even the only way to check that it works, since otherwise there's the auto-repair/regen.

It was tested on tremulous "bunker" and the behavior was the same.

I'm not sure to tag this bug or feature-request though.

**Comment by necessarily-equal** [comment]
I don't think this is the only way to test that it works, but for a start, did you see it doing anything on a buildable at all?

**Comment by bmorel** [comment]
no, because with OM/RC buildables just auto repair so can't know and had to remove power

**Comment by cu-kai** [comment]
specifically, tested with this QVM on Tremulous: https://github.com/enneract/AA-QVM/tree/staging

the buildables do not heal in the trigger_heal (with OM/RC destroyed) either on Unvanquished or Tremulous.

**Comment by bmorel** [comment]
Somewhat related to #1544 or at least referenced issue should be kept in mind if trigger_heal's behavior is to be updated to affect structures.

**Comment by bmorel** [comment]
I think trigger_heal and trigger_hurt should both affect structures, including when said structures are not finished to build yet.
Maybe an entity flag could be added to enable this behavior, or just a cvar? I'm not sure what's the best way to follow here: should mappers be able to tune this, or should server owners be allowed to do so? Maybe both?

**Comment by DolceTriade** [comment]
I think entity flags are a reasonable thing to add. We shouldn't break backwards compat with maps that didn't do this.

**Comment by bmorel** [comment]
I'm honestly tempted to use *both* as I think this is mostly harmless. The only condition where I see this can be a problem are the mission maps: some them have this ultra boring phase where aliens have to build over trigger_hurt bands or be able to do pixel-perfect jumps. I'm not saying those maps should be broken (certainly not!) but allowing to override the map's behavior on this point by exposing a cvar does not seems stupid to me.

**Comment by DolceTriade** [comment]
I really hate the idea of controlling per-map behavior with cvars. A server host should not have to have a config file to play a single map...

**Comment by cu-kai** [comment]
Also good point @DolceTriade, perhaps we should not have a cvar, and only an entity flag.
